### ISSUE TITLE
Setting/updating graph data onNodeDrag causing error

### ISSUE THREAD

**Issue by chancesmith** [vasturiano/force-graph#71]
**Describe the bug**
Setting graph data onNodeDrag causes an error: 
![Screenshot 2019-10-17 11 12 42](https://user-images.githubusercontent.com/4466585/67027383-10aa3f00-f0cf-11e9-9fe9-a6c6f2b19455.png)

**To Reproduce**
1. Update graphData with its current data onNodeDrag like in this function:
```
onNodeDrag: node => {
	let data = Graph.graphData()
	Graph.graphData(data)
},
``` 
2. Drag a node around
3. Check console for errors

**Expected behavior**
I expect to get the data, modify it, and return it without errors. It seems that just setting the data to its current value causes errors.

**Desktop (please complete the following information):**
 - OS: Mac OS Mojave 10.14.5
 - Browser chrome 78.0.3904.44
 - Force Graph version 1.21.9

**Comment by vasturiano** [comment]
@chancesmith thanks for reporting. I've tried to reproduce your issue in this Codepen: https://codepen.io/vasturiano/pen/ZEEeeyQ

But I don't find any errors. Perhaps the error is elsewhere. Could you make a minimal example that reproduces your issue?

**Comment by SZBoatwright** [comment]
@vasturiano I modified your codepen example a bit to have the issue show up faster. I changed N to 200 and changed onNodeDrag to onEngineTick. If you wait for a bit, you'll eventually see the error show up in the browser console. (it doesn't show in the codepen console for me)

https://codepen.io/SZBoatwright/pen/QWWajJj

**Comment by vasturiano** [comment]
Thanks for reporting this @SZBoatwright. It drew my attention to an internal memory leak bug related to how the nodes/links objects are tracked across data updates.

Should now be fixed in `v1.22.2`!
